cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1481,6 +1481,7 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 }
 
 func generateMountsModeRun(mst *initramfsMountsState) error {
```

Comment:
how much repetition is involved in having a generateMountsModeRunClassic? I think this was discussed a bit, I just wonder once we have more divergence whether that's better? a yet different approach is to have a behavior interface and two implementations that capture the core vs classic difference